app/locustio/common_utils.py

```diff
@@ -288,4 +292,38 @@ def raise_if_login_failed(locust):
         raise exception.StopUser('Action login_and_view_dashboard failed')
 
 
+def run_as_specific_user(username=None, password=None):
+    if not (username and password):
+        raise SystemExit(f'The credentials are not valid: {{username: {username}, password: {password}}}.')
+
+    def deco_wrapper(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+
+            locust = None
+            for obj in list(locals()['kwargs'].values()) + list(locals()['args']):
+                if isinstance(obj, MyBaseTaskSet):
+                    locust = obj
+                    break
+
+            if locust:
+                session_user_name = locust.session_data_storage["username"]
+                session_user_password = locust.session_data_storage["password"]
+                url = JIRA_API_URL if locust.session_data_storage['app'] == 'jira' or 'jsm' else CONFLUENCE_API_URL
```

Comment:
Need to handle case when app type is not known.

Reply:
Done